homeassistant/config_entries.py

```diff
@@ -1211,7 +1211,10 @@ def _async_abort_entries_match(
         if match_dict is None:
             match_dict = {}  # Match any entry
         for entry in self._async_current_entries(include_ignore=False):
-            if all(item in entry.data.items() for item in match_dict.items()):
+            if all(
+                item in {**entry.data, **entry.options}.items()
```

Comment:
There may be conflicting keys in those dicts?
Might it be safer to merge the `entry.data.items()` and `entry.options.items()` lists instead?

Reply:
For conflicting keys, one could decide to create a set of both using `entry.data.items() | entry.options.items()` but I think that might be even more confusing.

Reply:
Thinking here, wouldn't it make more sense to just pass a 2nd dictionary to `_async_abort_entries_match` that matches the options ?

Reply:
> you are now creating a new dict for every key in `match_dict`.
> 
> Let's move creation above the if-check. You could also decide to use `from itertools import chain`.

Yeah thats inefficient, I switch it to use `ChainMap`

Reply:
Yeah I agree on the latter, more specific

Reply:
> Thinking here, wouldn't it make more sense to just pass a 2nd dictionary to `_async_abort_entries_match` that matches the options ?

We are matching data against existing config entries.  The input we have is the data not the config entries so I don't see how that would work

Reply:
Checked all the existing usage. Its almost all `CONF_USERNAME`, `CONF_HOST`, `CONF_URL`, `CONF_IP_ADDRESS`, `CONF_DEVICE`.  If its present in both data and options that probably a bug in the integration so I think its probably not likely to happen in reality.  I don't think it is a likely (or valid) use case to store different values for these type of keys in options and data, especially if we are checking to see if we should abort if its present.